features/ticket/ticket-create.ts

```diff
@@ -0,0 +1,129 @@
+import {
+    MessageEmbed,
+    MessageActionRow,
+    MessageButton,
+    Client,
+} from "discord.js";
+
+import TicketConfigSchema from "../../models/ticketconfig-schema";
+import TicketSystemSchema from "../../models/ticketsystem-schema";
+
+export default (client: Client) => {
+    client.on("interactionCreate", async (ButtonInteraction) => {
+        if (!ButtonInteraction.isButton()) return;
+        const { guild, member, customId } = ButtonInteraction;
+
+        if (!guild) return;
+        if (!member) return;
+
+        const Data = await TicketConfigSchema.findOne({ GuildId: guild.id });
+        if (!Data) return;
+
+        if (!Data.Buttons.includes(customId)) return;
+
+        let TicketCount = Data.GuildTicketCount;
+
+        ButtonInteraction.reply({
+            content: "creating....",
+            ephemeral: true,
+        });
+
+        await guild.channels
+            .create(`${customId}-Ticket-${TicketCount}`, {
+                type: "GUILD_TEXT",
+                reason: `Ticket Created by: ${member.user.username} | id= ${member.user.id}`,
+                parent: Data.OpenCategoryID,
+                permissionOverwrites: [
+                    {
+                        id: member.user.id,
+                        allow: [
+                            "SEND_MESSAGES",
+                            "VIEW_CHANNEL",
+                            "READ_MESSAGE_HISTORY",
+                            "ATTACH_FILES",
+                        ],
+                    },
+                    {
+                        id: Data.EveryoneRoleID,
+                        deny: ["VIEW_CHANNEL", "SEND_MESSAGES"],
+                    },
+                    {
+                        id: Data.SupportRoleID,
+                        allow: [
+                            "SEND_MESSAGES",
+                            "VIEW_CHANNEL",
+                            "READ_MESSAGE_HISTORY",
+                            "ATTACH_FILES",
+                            "MANAGE_MESSAGES",
+                        ],
+                    },
+                ],
+            })
+            .then(async (channel) => {
+                ButtonInteraction.followUp({

```

Comment:
brooo thus Followup reply never works :/ .... i am so dumb cause can't event identify the error